api/src/controllers/userController.js

```
@@ -48,3 +47,65 @@ export const getUserApplicationsById = async (req, res) => {
     return res.status(500).json({ error: 'Internal Server Error' })
   }
 }
+export const postUserApplications = async (req, res) => {
+  try {
+    const { appData, user_id } = req.body
+
+    // Check if the company name already exists
+    const existingCompany = await knex('company')
+      .where({ name: appData.company_name })
+      .first()
+
+    let company_id
+    if (existingCompany) {
+      company_id = existingCompany.company_id
+    } else {
+      // Insert the company and fetch the company ID immediately
+
+      const [newCompany] = await knex('company')
+        .insert({
+          name: appData.company_name,
+        })
+        .returning('company_id') // Get the inserted company's ID
+      company_id = newCompany.company_id
+    }
+    // Insert the data for the new application
+    await knex('application').insert({
+      user_id: user_id,
```

You are not validating that user_id mentioned here is same as user_id in the JWT. This means user1 can put id of user2 if they were spamming your API endpoints directly.

what about user_id being wrong? Foreign key constraint will fail on DB level, so you should validate the inputs first before continuing. Fail early when handling a request vs blowing up in the end with a very long try/catch block.

---

Yeah that is totally make sense. I made the changes now, I get the user ID from the token instead of sending from frontend. Because when I am making post request I already send the token along with it in header for validation. 

I hope the solution make sense :)